### PR TITLE
upgrade docker images to ROCm 5.0.2

### PR DESCRIPTION
**Pull request by causten**
Move to CI containers to rocm 5.0.2

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,7 +6,8 @@ ARG PREFIX=/usr/local
 RUN dpkg --add-architecture i386
 
 # Add rocm repository
-RUN sh -c 'echo deb [arch=amd64 trusted=yes] http://repo.radeon.com/rocm/apt/4.5/ ubuntu main > /etc/apt/sources.list.d/rocm.list'
+RUN sh -c 'echo deb [arch=amd64 trusted=yes] http://repo.radeon.com/rocm/apt/5.0.2/ ubuntu main > /etc/apt/sources.list.d/rocm.list'
+RUN sh -c 'echo deb [arch=amd64 trusted=yes] http://repo.radeon.com/amdgpu/21.50.2/ubuntu bionic main > /etc/apt/sources.list.d/amdgpu.list'
```
Comment (pfultz2) [inline Dockerfile:10]: Is this needed? We only install the userspace packages in the docker, and I think this doesn't have any userspace packages.

**Comment by pfultz2** [comment]
Besides fixing the compiler warnings, the CI nodes need to be upgraded to rocm 5.0 before we can merge this in.

File: .clang-tidy [hunk 2]
```diff
@@ -113,3 +113,5 @@ CheckOptions:
     value:           UPPER_CASE
   - key:             readability-identifier-naming.MacroDefinitionPrefix
     value:           MIGRAPHX_
+  - key:             readability-identifier-length.MinimumVariableNameLength
```
Comment (pfultz2) [inline .clang-tidy:116]: I think it would be better to disable this check instead.

Comment (causten) [reply .clang-tidy:116]: Spec says to disable set to a 0 or a 1.

Comment (pfultz2) [reply .clang-tidy]: To disable a check you a put `-` in front of the name in the list [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/CMakeLists.txt#L149).

File: Dockerfile (hunk 1 above)
Comment (causten) [reply Dockerfile]: removed the amdgpu installer repo

File: .clang-tidy (hunk 2 above)
Comment (causten) [reply .clang-tidy]: Done

**Review by kahmed10 — APPROVED** [review]
Someone can either make a PR for #1176 or commit changes here and merge this in?

File: CMakeLists.txt [hunk 3]
```diff
@@ -93,11 +93,14 @@ rocm_enable_clang_tidy(
         modernize-*
         performance-*
         readability-*
-        -bugprone-signed-char-misuse
+        -bugprone-easily-swappable-parameters
+	-bugprone-implicit-widening-of-multiplication-result
```
Comment (pfultz2) [inline CMakeLists.txt:97]: This is not formatted correctly.

File: cmake/EnableCompilerWarnings.cmake [hunk 4]
```diff
@@ -96,12 +96,15 @@ else()
                 -Wno-gnu-zero-variadic-macro-arguments
                 -Wno-missing-prototypes
                 -Wno-nested-anon-types
+		-Wno-option-ignored
```
Comment (pfultz2) [inline cmake/EnableCompilerWarnings.cmake:99]: This should be formatted.

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -15,7 +15,9 @@ jobs:
 
     steps:
     - name: Free space
-      run: sudo rm -rf /usr/local/android /usr/share/dotnet /usr/local/share/boost /opt/ghc /usr/local/share/chrom* /usr/share/swift /usr/local/julia*
+      run: |
+              sudo rm -rf /usr/local/android /usr/share/dotnet /usr/local/share/boost /opt/ghc /usr/local/share/chrom* /usr/share/swift /usr/local/julia* /usr/local/lib/android
```
Comment (pfultz2) [inline .github/workflows/ci.yaml:19]: This line should be the same as the other lines.

File: cmake/EnableCompilerWarnings.cmake [hunk 6]
```diff
@@ -96,12 +96,15 @@ else()
...
                 -Wno-padded
                 -Wno-shorten-64-to-32
                 -Wno-sign-conversion
                 -Wno-unused-command-line-argument
                 -Wno-weak-vtables
                 -Wno-c99-extensions
+                -Wno-reserved-macro-identifier
+                -Wno-reserved-identifier
```
Comment (pfultz2) [inline cmake/EnableCompilerWarnings.cmake:107]: These warnings should not be disabled.

File: CMakeLists.txt (hunk 3 above)
Comment (causten) [reply CMakeLists.txt:97]: Fixed

File: cmake/EnableCompilerWarnings.cmake (hunk 4 above)
Comment (causten) [reply cmake/EnableCompilerWarnings.cmake:99]: fixed

File: cmake/EnableCompilerWarnings.cmake (hunk 6 above)
Comment (causten) [reply cmake/EnableCompilerWarnings.cmake:107]: fixed

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**